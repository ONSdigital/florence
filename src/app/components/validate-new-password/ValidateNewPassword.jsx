import React, { Component } from "react";
import Panel from "../panel/Panel";
import PropTypes from "prop-types";
import Input from "../Input";
import { errCodes } from "../../utilities/errorCodes";
import notifications from "../../utilities/notifications";
import ValidationItemList from "../../components/validation-item-list/ValidationItemList";

const propTypes = {
    updateValidity: PropTypes.func
};

export class ValidateNewPassword extends Component {
    constructor(props) {
        super(props);
        this.state = {
            password: {
                value: "",
                type: "password"
            },
            validationRules: [
                {
                    name: "14 characters",
                    checked: false,
                    id: "minimum-character-limit"
                },
                {
                    name: "1 uppercase character",
                    checked: false,
                    id: "uppercase-character-validation"
                },
                {
                    name: "1 lowercase character",
                    checked: false,
                    id: "lowercase-character-validation"
                },
                {
                    name: "1 number",
                    checked: false,
                    id: "minimum-number-limit"
                }
            ]
        };

        this.handleInputChange = this.handleInputChange.bind(this);
    }

    checkPasswordValidation() {
        this.setState(
            {
                validationRules: [
                    {
                        name: "14 characters",
                        checked: this.checkStringLength(),
                        id: "minimum-character-limit"
                    },
                    {
                        name: "1 uppercase character",
                        checked: this.checkUpperCase(),
                        id: "uppercase-character-validation"
                    },
                    {
                        name: "1 lowercase character",
                        checked: this.checkLowerCase(),
                        id: "lowercase-character-validation"
                    },
                    {
                        name: "1 number",
                        checked: this.checkNumberPresence(),
                        id: "minimum-number-limit"
                    }
                ]
            },

            () => {
                let ruleChecked = true;
                this.state.validationRules.map(rule => {
                    if (!rule.checked) {
                        ruleChecked = false;
                    }
                });
                this.props.updateValidity(ruleChecked, this.state.password.value);
            }
        );
    }

    checkStringLength() {
        const charMinLength = 13;
        return this.state.password.value.length > charMinLength;
    }

    checkUpperCase() {
        return /^.*[A-Z].*$/.test(this.state.password.value);
    }

    checkLowerCase() {
        return /^.*[a-z].*$/.test(this.state.password.value);
    }

    checkNumberPresence() {
        return /^.*[0-9].*$/.test(this.state.password.value);
    }

    handleInputChange(event) {
        const id = event.target.id;
        const value = event.target.value;
        const checked = event.target.checked;

        switch (id) {
            case "password-input": {
                this.setState(
                    {
                        password: {
                            ...this.state.password,
                            value: value
                        }
                    },
                    this.checkPasswordValidation
                );
                break;
            }
            case "password-checkbox": {
                this.setState(prevState => ({
                    password: {
                        ...prevState.password,
                        type: checked ? "text" : "password"
                    }
                }));
                break;
            }
            default: {
                const notification = {
                    type: "warning",
                    isDismissable: true,
                    autoDismiss: 15000
                };
                console.error(errCodes.UNEXPECTED_ERR);
                notification.message = errCodes.UNEXPECTED_ERR;
                notifications.add(notification);
            }
        }
    }

    render() {
        const validateNewPasswordPanelBody = <ValidationItemList validationRules={this.state.validationRules} />;
        return (
            <div>
                <div className="margin-bottom--1">
                    <Panel type="information" heading="Your password must have at least:" body={validateNewPasswordPanelBody} />
                </div>
                <Input
                    id="password-input"
                    type={this.state.password.type}
                    label="Password"
                    onChange={this.handleInputChange}
                    disableShowPasswordText={true}
                    value={this.state.password.value}
                />
                <Input
                    id="password-checkbox"
                    type="checkbox"
                    label="Show password"
                    onChange={this.handleInputChange}
                    reverseLabelOrder={true}
                    inline={true}
                />
            </div>
        );
    }
}

ValidateNewPassword.propTypes = propTypes;
export default ValidateNewPassword;
