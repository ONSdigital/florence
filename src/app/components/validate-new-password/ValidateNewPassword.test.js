import React from "react";
import { mount } from "enzyme";
import ValidateNewPassword from "./ValidateNewPassword";
import renderer from "react-test-renderer";

let updateValidity = jest.fn(isValid => {
    // do nothing
});

const props = {
    updateValidity: updateValidity
};
describe("When the validate new password component is added", () => {
    const component = mount(<ValidateNewPassword {...props} />);
    it("have an initial state with no password typed yet", () => {
        expect(component.state().password.value).toBe("");
    });
    it("should have a character limit validation checkbox which is unchecked", () => {
        const minCharCheckbox = component.find("#minimum-character-limit");
        expect(minCharCheckbox.exists()).toBe(true);
        expect(minCharCheckbox.props()["checked"]).toBe(false);
        expect(component.state().validationRules[0].checked).toBe(false);
    });
    it("should have a required uppercase character validation checkbox which is unchecked", () => {
        const upperCharCheckbox = component.find("#uppercase-character-validation");
        expect(upperCharCheckbox.exists()).toBe(true);
        expect(upperCharCheckbox.props()["checked"]).toBe(false);
        expect(component.state().validationRules[1].checked).toBe(false);
    });
    it("should have a required lowercase character validation checkbox which is unchecked", () => {
        const lowerCharCheckbox = component.find("#lowercase-character-validation");
        expect(lowerCharCheckbox.exists()).toBe(true);
        expect(lowerCharCheckbox.props()["checked"]).toBe(false);
        expect(component.state().validationRules[2].checked).toBe(false);
    });
    it("should have a required numerical character validation checkbox which is unchecked", () => {
        const numCharCheckbox = component.find("#minimum-number-limit");
        expect(numCharCheckbox.exists()).toBe(true);
        expect(numCharCheckbox.props()["checked"]).toBe(false);
        expect(component.state().validationRules[3].checked).toBe(false);
    });
    it("should have an input field", () => {
        // Check it exists
        const inputBox = component.find("#password-input");
        expect(inputBox.exists()).toBe(true);
        expect(inputBox.prop("value")).toBe("");
        expect(component.find("#password-checkbox").exists()).toBe(true);
    });
});

describe("When a lowercase character is entered into the text input field", () => {
    const component = mount(<ValidateNewPassword {...props} />);
    const numCharCheckbox = component.find("#minimum-number-limit");
    const upperCharCheckbox = component.find("#uppercase-character-validation");
    const lowerCharCheckbox = component.find("#lowercase-character-validation");
    const minCharCheckbox = component.find("#minimum-character-limit");
    component.instance().handleInputChange({
        target: {
            id: "password-input",
            value: "a"
        }
    });
    it("Should update the validation rule states", () => {
        expect(component.state().validationRules[0].checked).toBe(false);
        expect(component.state().validationRules[1].checked).toBe(false);
        expect(component.state().validationRules[2].checked).toBe(true);
        expect(component.state().validationRules[3].checked).toBe(false);
    });
    it("Should pass validation for lowercase character required", () => {
        expect(numCharCheckbox.props()["checked"]).toBe(false);
        expect(upperCharCheckbox.props()["checked"]).toBe(false);
        expect(lowerCharCheckbox.props()["checked"]).toBe(true);
        expect(minCharCheckbox.props()["checked"]).toBe(false);
    });
    it("Should fail overall validation check", () => {
        expect(component.props().updateValidity).toHaveBeenCalledWith(false);
    });
    it("should match the snapshot", () => {
        const props = {
            override: true
        };
        const component = renderer.create(<ValidateNewPassword {...props} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
describe("When an uppercase character is entered into the text input field", () => {
    const component = mount(<ValidateNewPassword {...props} />);
    const numCharCheckbox = component.find("#minimum-number-limit");
    const upperCharCheckbox = component.find("#uppercase-character-validation");
    const lowerCharCheckbox = component.find("#lowercase-character-validation");
    const minCharCheckbox = component.find("#minimum-character-limit");
    component.instance().handleInputChange({
        target: {
            id: "password-input",
            value: "A"
        }
    });
    it("Should update the validation rule states", () => {
        expect(component.state().validationRules[0].checked).toBe(false);
        expect(component.state().validationRules[1].checked).toBe(true);
        expect(component.state().validationRules[2].checked).toBe(false);
        expect(component.state().validationRules[3].checked).toBe(false);
    });
    it("Should pass validation for Upper Case character required", () => {
        expect(numCharCheckbox.props()["checked"]).toBe(false);
        expect(upperCharCheckbox.props()["checked"]).toBe(true);
        expect(lowerCharCheckbox.props()["checked"]).toBe(false);
        expect(minCharCheckbox.props()["checked"]).toBe(false);
    });
    it("Should fail overall validation check", () => {
        expect(component.props().updateValidity).toHaveBeenCalledWith(false);
    });
    it("should match the snapshot", () => {
        const props = {
            override: true
        };
        const component = renderer.create(<ValidateNewPassword {...props} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
describe("When a numerical character is entered into the text input field", () => {
    const component = mount(<ValidateNewPassword {...props} />);
    const numCharCheckbox = component.find("#minimum-number-limit");
    const upperCharCheckbox = component.find("#uppercase-character-validation");
    const lowerCharCheckbox = component.find("#lowercase-character-validation");
    const minCharCheckbox = component.find("#minimum-character-limit");
    component.instance().handleInputChange({
        target: {
            id: "password-input",
            value: "1"
        }
    });
    it("Should update the validation rule states", () => {
        expect(component.state().validationRules[0].checked).toBe(false);
        expect(component.state().validationRules[1].checked).toBe(false);
        expect(component.state().validationRules[2].checked).toBe(false);
        expect(component.state().validationRules[3].checked).toBe(true);
    });
    it("Should pass validation for numerical character required", () => {
        expect(numCharCheckbox.props()["checked"]).toBe(true);
        expect(upperCharCheckbox.props()["checked"]).toBe(false);
        expect(lowerCharCheckbox.props()["checked"]).toBe(false);
        expect(minCharCheckbox.props()["checked"]).toBe(false);
    });
    it("Should fail overall validation check", () => {
        expect(component.props().updateValidity).toHaveBeenCalledWith(false);
    });
    it("should match the snapshot", () => {
        const props = {
            override: true
        };
        const component = renderer.create(<ValidateNewPassword {...props} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});

describe("When fourteen numerical and upper as well as lower characters are entered into the text input field", () => {
    const component = mount(<ValidateNewPassword {...props} />);
    const numCharCheckbox = component.find("#minimum-number-limit");
    const upperCharCheckbox = component.find("#uppercase-character-validation");
    const lowerCharCheckbox = component.find("#lowercase-character-validation");
    const minCharCheckbox = component.find("#minimum-character-limit");
    component.instance().handleInputChange({
        target: {
            id: "password-input",
            value: "aB1cD2eF3gH4iJ"
        }
    });
    it("Should update the validation rule states", () => {
        // Check that the state has been updated
        expect(component.state().password.value).toBe("aB1cD2eF3gH4iJ");
        expect(component.state().validationRules[0].checked).toBe(true);
        expect(component.state().validationRules[1].checked).toBe(true);
        expect(component.state().validationRules[2].checked).toBe(true);
        expect(component.state().validationRules[3].checked).toBe(true);
    });
    it("Should pass validation for minimum character limit required and all others", () => {
        // Check that the character limit requirement checkbox is now checked
        expect(numCharCheckbox.props()["checked"]).toBe(true);
        expect(upperCharCheckbox.props()["checked"]).toBe(true);
        expect(lowerCharCheckbox.props()["checked"]).toBe(true);
        expect(minCharCheckbox.props()["checked"]).toBe(true);
    });
    it("Should pass overall validation check", () => {
        expect(component.props().updateValidity).toHaveBeenCalledWith(true);
    });
    it("should match the snapshot", () => {
        const props = {
            override: true
        };
        const component = renderer.create(<ValidateNewPassword {...props} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
