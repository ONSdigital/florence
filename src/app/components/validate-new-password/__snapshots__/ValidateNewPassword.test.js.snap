// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When a lowercase character is entered into the text input field should match the snapshot 1`] = `
<div>
  <div
    className="margin-bottom--1"
  >
    <div
      className="information"
    >
      <div
        className=""
      >
        <div
          className="panel panel--information panel--no-title"
        >
          <span
            className="visually-hidden"
          >
            information
          </span>
          <div
            className="panel__body"
          >
            <p
              className="panel__information"
            >
              <strong
                className="font-weight--600"
              >
                Your password must have at least:
              </strong>
            </p>
            <div
              className="margin-bottom--1"
            >
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="minimum-character-limit"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="minimum-character-limit"
                >
                   
                  14 characters
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="uppercase-character-validation"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="uppercase-character-validation"
                >
                   
                  1 uppercase character
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="lowercase-character-validation"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="lowercase-character-validation"
                >
                   
                  1 lowercase character
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="minimum-number-limit"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="minimum-number-limit"
                >
                   
                  1 number
                </label>
              </div>
              <p
                className="margin-top--1 margin-bottom--1"
              >
                It could be a phrase that has more than 14 characters. For example, 1bought11cupsoftea.
              </p>
              <p>
                It can also have special characters, for example £,?,!,%,&.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form__input"
  >
    <label
      className="form__label"
      htmlFor="password-input"
    >
      Password
    </label>
    
    <input
      autoComplete="new-password"
      autoFocus={false}
      className="input"
      disabled={false}
      id="password-input"
      name="password-input"
      onChange={[Function]}
      onFocus={[Function]}
      type="password"
      value=""
    />
    
  </div>
  <div
    className="form__input form__input--flush checkbox"
  >
    
    <input
      autoComplete="new-password"
      autoFocus={false}
      className="input checkbox__input visually-hidden"
      disabled={false}
      id="password-checkbox"
      name="password-checkbox"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Show password"
      type="checkbox"
    />
    
    <label
      className="form__label checkbox__label font-weight--300"
      htmlFor="password-checkbox"
    >
      Show password
    </label>
  </div>
</div>
`;

exports[`When a numerical character is entered into the text input field should match the snapshot 1`] = `
<div>
  <div
    className="margin-bottom--1"
  >
    <div
      className="information"
    >
      <div
        className=""
      >
        <div
          className="panel panel--information panel--no-title"
        >
          <span
            className="visually-hidden"
          >
            information
          </span>
          <div
            className="panel__body"
          >
            <p
              className="panel__information"
            >
              <strong
                className="font-weight--600"
              >
                Your password must have at least:
              </strong>
            </p>
            <div
              className="margin-bottom--1"
            >
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="minimum-character-limit"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="minimum-character-limit"
                >
                   
                  14 characters
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="uppercase-character-validation"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="uppercase-character-validation"
                >
                   
                  1 uppercase character
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="lowercase-character-validation"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="lowercase-character-validation"
                >
                   
                  1 lowercase character
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="minimum-number-limit"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="minimum-number-limit"
                >
                   
                  1 number
                </label>
              </div>
              <p
                className="margin-top--1 margin-bottom--1"
              >
                It could be a phrase that has more than 14 characters. For example, 1bought11cupsoftea.
              </p>
              <p>
                It can also have special characters, for example £,?,!,%,&.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form__input"
  >
    <label
      className="form__label"
      htmlFor="password-input"
    >
      Password
    </label>
    
    <input
      autoComplete="new-password"
      autoFocus={false}
      className="input"
      disabled={false}
      id="password-input"
      name="password-input"
      onChange={[Function]}
      onFocus={[Function]}
      type="password"
      value=""
    />
    
  </div>
  <div
    className="form__input form__input--flush checkbox"
  >
    
    <input
      autoComplete="new-password"
      autoFocus={false}
      className="input checkbox__input visually-hidden"
      disabled={false}
      id="password-checkbox"
      name="password-checkbox"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Show password"
      type="checkbox"
    />
    
    <label
      className="form__label checkbox__label font-weight--300"
      htmlFor="password-checkbox"
    >
      Show password
    </label>
  </div>
</div>
`;

exports[`When an uppercase character is entered into the text input field should match the snapshot 1`] = `
<div>
  <div
    className="margin-bottom--1"
  >
    <div
      className="information"
    >
      <div
        className=""
      >
        <div
          className="panel panel--information panel--no-title"
        >
          <span
            className="visually-hidden"
          >
            information
          </span>
          <div
            className="panel__body"
          >
            <p
              className="panel__information"
            >
              <strong
                className="font-weight--600"
              >
                Your password must have at least:
              </strong>
            </p>
            <div
              className="margin-bottom--1"
            >
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="minimum-character-limit"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="minimum-character-limit"
                >
                   
                  14 characters
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="uppercase-character-validation"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="uppercase-character-validation"
                >
                   
                  1 uppercase character
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="lowercase-character-validation"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="lowercase-character-validation"
                >
                   
                  1 lowercase character
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="minimum-number-limit"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="minimum-number-limit"
                >
                   
                  1 number
                </label>
              </div>
              <p
                className="margin-top--1 margin-bottom--1"
              >
                It could be a phrase that has more than 14 characters. For example, 1bought11cupsoftea.
              </p>
              <p>
                It can also have special characters, for example £,?,!,%,&.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form__input"
  >
    <label
      className="form__label"
      htmlFor="password-input"
    >
      Password
    </label>
    
    <input
      autoComplete="new-password"
      autoFocus={false}
      className="input"
      disabled={false}
      id="password-input"
      name="password-input"
      onChange={[Function]}
      onFocus={[Function]}
      type="password"
      value=""
    />
    
  </div>
  <div
    className="form__input form__input--flush checkbox"
  >
    
    <input
      autoComplete="new-password"
      autoFocus={false}
      className="input checkbox__input visually-hidden"
      disabled={false}
      id="password-checkbox"
      name="password-checkbox"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Show password"
      type="checkbox"
    />
    
    <label
      className="form__label checkbox__label font-weight--300"
      htmlFor="password-checkbox"
    >
      Show password
    </label>
  </div>
</div>
`;

exports[`When fourteen numerical and upper as well as lower characters are entered into the text input field should match the snapshot 1`] = `
<div>
  <div
    className="margin-bottom--1"
  >
    <div
      className="information"
    >
      <div
        className=""
      >
        <div
          className="panel panel--information panel--no-title"
        >
          <span
            className="visually-hidden"
          >
            information
          </span>
          <div
            className="panel__body"
          >
            <p
              className="panel__information"
            >
              <strong
                className="font-weight--600"
              >
                Your password must have at least:
              </strong>
            </p>
            <div
              className="margin-bottom--1"
            >
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="minimum-character-limit"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="minimum-character-limit"
                >
                   
                  14 characters
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="uppercase-character-validation"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="uppercase-character-validation"
                >
                   
                  1 uppercase character
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="lowercase-character-validation"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="lowercase-character-validation"
                >
                   
                  1 lowercase character
                </label>
              </div>
              <div
                className="readonly-checkbox__wrapper"
              >
                <input
                  checked={false}
                  className="readonly-checkbox"
                  id="minimum-number-limit"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="checkbox"
                />
                <label
                  htmlFor="minimum-number-limit"
                >
                   
                  1 number
                </label>
              </div>
              <p
                className="margin-top--1 margin-bottom--1"
              >
                It could be a phrase that has more than 14 characters. For example, 1bought11cupsoftea.
              </p>
              <p>
                It can also have special characters, for example £,?,!,%,&.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form__input"
  >
    <label
      className="form__label"
      htmlFor="password-input"
    >
      Password
    </label>
    
    <input
      autoComplete="new-password"
      autoFocus={false}
      className="input"
      disabled={false}
      id="password-input"
      name="password-input"
      onChange={[Function]}
      onFocus={[Function]}
      type="password"
      value=""
    />
    
  </div>
  <div
    className="form__input form__input--flush checkbox"
  >
    
    <input
      autoComplete="new-password"
      autoFocus={false}
      className="input checkbox__input visually-hidden"
      disabled={false}
      id="password-checkbox"
      name="password-checkbox"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Show password"
      type="checkbox"
    />
    
    <label
      className="form__label checkbox__label font-weight--300"
      htmlFor="password-checkbox"
    >
      Show password
    </label>
  </div>
</div>
`;
